File: chingoon.c
#include "stdlib.h"
#include "stdio.h"
#define top_head " __________"
#define eyes     "(--[ .]-[ .]"
#define fatty    "(       O  )"
#define slim     "(_______O__)"
#define fat    "(          )"
#define chin     "(__________)"

void print_chin_goon(int chins) {
  int i = 0;
  puts(top_head);
  puts(eyes);
  if (chins > 0) {
    puts(fatty);
    for (i = 1 ; i < chins ; i++ ) {
      puts(fat);
    }
    puts(chin);
  } else if (chins <= 0) {
    puts(slim);
  }
}

int main(int argc, char ** argv) {
  int chins = 0;
  if (argc <= 1) {
    puts("Need an argument!");
    return 1;
  }
  chins = atoi(argv[1]);
  print_chin_goon(chins);
  return 0;
}

File: chingoon.lua
#!/usr/bin/env lua

-- face part table
face_parts = {
  TOP   = ' __________',
  EYES  = '(--[ .]-[ .]',
  FATTY = '(       O  )',
  SLIM  = '(_______O__)',
  FAT   = '(          )',
  CHIN  = '(__________)'
}

-- validate chin arg, if there is one. else, default to 0
if arg[1] == nil then
  chins = 0
else
  chins = assert(tonumber(arg[1]), "Chins has to be a number")
end

-- disallow negative chin count
if chins < 0 then error("Can't have a negachin") end

-- begin printing the face
print(face_parts.TOP)
print(face_parts.EYES)

-- slim
if chins == 0 then
  print("(_______O__)")
  return
end

-- fatty
fatprinter = coroutine.create(
  function(c)
    print(face_parts.FATTY)
    for i=1, chins do
      print(face_parts.FAT)
    end
    print(face_parts.CHIN)
  end)

coroutine.resume(fatprinter, chins)
